core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java

```diff
@@ -131,16 +179,18 @@ public OAuth2Config setClientSecret(String clientSecret)
         return this;
     }
 
-    public Optional<String> getAudience()
+    @NotNull
+    public List<String> getAdditionalAudiences()
     {
-        return audience;
+        return additionalAudiences;
     }
 
-    @Config("http-server.authentication.oauth2.audience")
-    @ConfigDescription("The required audience of a token")
-    public OAuth2Config setAudience(String audience)
+    @LegacyConfig("http-server.authentication.oauth2.audience")
```

Comment:
I missed this during review. this should be something like below (if the configs are indeed equivalent).

```java
    @LegacyConfig(value = "http-server.authentication.oauth2.audience", replacedBy = "http-server.authentication.oauth2.additional-audiences")
```

Otherwise old config files will cause Trino startup to fail with an error like:

```
io.airlift.bootstrap.ApplicationConfigurationException: 
Configuration errors:

1) Error: Invalid configuration property http-server.authentication.oauth2.issuer: may not be null (for class OAuth2Config.issuer)

2) Error: Configuration property 'http-server.authentication.oauth2.audience' has been replaced. Use 'http-server.authentication.oauth2.additional-audiences' instead.
```

This should also be mentioned in the release notes and documentation. cc: @11xor6 @kokosing

Reply:
If the configs are not interchangeable then the existing code looks good to me.